lib/ansible/plugins/action/reboot.py

```diff
@@ -130,21 +131,39 @@ def get_distribution(self, task_vars):
 
     def get_shutdown_command(self, task_vars, distribution):
         shutdown_bin = self._get_value_from_facts('SHUTDOWN_COMMANDS', distribution, 'DEFAULT_SHUTDOWN_COMMAND')
-
-        display.debug('{action}: running find module to get path for "{command}"'.format(action=self._task.action, command=shutdown_bin))
+        default_search_paths = ['/sbin', '/usr/sbin', '/usr/local/sbin']
+        search_paths = self._task.args.get('search_paths', default_search_paths)
+
+        # Convert bare strings to a list
+        if is_string(search_paths):
+            search_paths = [search_paths]
+
+        # Error if we didn't get a list
+        if not isinstance(search_paths, list):
+            raise AnsibleError("'search_paths' must be a string or list, got {0}: {1}".format(type(search_paths), search_paths))
+
+        # If we did get a list, ensure it's a flast list and does not contain nested data structures
+        nested_list = any(isinstance(x, (dict, list)) for x in search_paths)
```

Comment:
should probably be `all(is_string(x ...` instead; if the path isn't a string, it's a fail, right?

Reply:
This check is looking for lists or dicts in the `search_paths` field:

```
search_paths:
  -
    - /some/path
    - /some/other/path
  hey: there
```

I can add another check to each of the search paths are actually strings.